src/app/pokemons/store/reducers/pokemons.reducer.ts

```diff
@@ -0,0 +1,105 @@
+import {createEntityAdapter, EntityAdapter, EntityState} from '@ngrx/entity';
+import {Pokemon} from '../../../models/pokemon';
+import {Action, createReducer, on} from '@ngrx/store';
+import {environment} from '../../../../environments/environment';
+import * as PokemonsActions from '../actions/pokemons.actions';
+import {AppError} from '../../../utils/error';
+
+export const pokemonsKey = 'pokemons';
+
+export interface PokemonsState extends EntityState<Pokemon> {
+  nextUrl: string;
+  searchedPokemon: string;
+  comparing: boolean;
+  toCompare: Pokemon;
+  toShow: Pokemon;
+  error: AppError;
+}
+
+export const adapter: EntityAdapter<Pokemon> = createEntityAdapter<Pokemon>();
+
+export const initialState: PokemonsState = adapter.getInitialState({
+  nextUrl: environment.pokemonsUrl,
+  searchedPokemon: '',
+  comparing: false,
+  toCompare: null,
+  toShow: null,
+  error: null
+});
+
+const pokemonsReducer = createReducer(
+  initialState,
+  on(PokemonsActions.loadPokemonsSuccess,
+    (state, {nextUrl, pokemons}) => {
+      return adapter.setAll(pokemons, {...state, nextUrl});
+    }),
+  on(PokemonsActions.addPokemonsSuccess,
+    (state, {nextUrl, pokemons}) => {
+      localStorage.setItem('nextUrl', nextUrl);
```

Comment:
why are you storing the next url in localstorage?

Reply:
To save the pokemons and to have the real reference to the next pokemons to load